manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.21.yaml

```diff
@@ -1,486 +0,0 @@
-{{- if and .Values.telemetry.enabled .Values.telemetry.v2.enabled }}
-{{ $customProm := not (include "default-prometheus" . | eq "true") }}
-{{ $customSdMetrics := not (include "default-sd-metrics" . | eq "true") }}
-{{ $customSdLogs := not (include "default-sd-logs" . | eq "true") }}
----
-# Note: http stats filter is wasm enabled only in sidecars.
-{{- if and .Values.telemetry.v2.prometheus.enabled $customProm }}
-apiVersion: networking.istio.io/v1alpha3
-kind: EnvoyFilter
-metadata:
-  name: stats-filter-1.21{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
-  {{- if .Values.meshConfig.rootNamespace }}
-  namespace: {{ .Values.meshConfig.rootNamespace }}
-  {{- else }}
-  namespace: {{ .Release.Namespace }}
-  {{- end }}
-  labels:
-    istio.io/rev: {{ .Values.revision | default "default" }}
-spec:
-  priority: -1
-  configPatches:
-    - applyTo: HTTP_FILTER
-      match:
-        context: SIDECAR_OUTBOUND
-        proxy:
-          proxyVersion: '^1\.21.*'
-        listener:
-          filterChain:
-            filter:
-              name: "envoy.filters.network.http_connection_manager"
-              subFilter:
-                name: "envoy.filters.http.router"
-      patch:
-        operation: INSERT_BEFORE
-        value:
-          name: istio.stats
```

Comment:
These EnvoyFilters were already unused in 99% of scenarios and instead set via https://github.com/istio/istio/blob/cd981f42ba33497af1e5ec7b596838295e7f5355/manifests/charts/istio-control/istio-discovery/templates/configmap.yaml#L24 since 1.18. (fwiw on our MCP product, this has been the case since launch)

There is no change to behavior for almost all users. For the very very few users using audit log or access log policy (as discussed in the other thread), there is a slightly different enablement mechanism; the docs will be updated for this.